daemon/analytics.c

```diff
@@ -335,6 +357,76 @@ void analytics_alarms_notifications(void)
     buffer_free(b);
 }
 
+char *get_value_from_key(char *buffer, char *key)
+{
+    char *s = NULL, *t = NULL;
+    s = t = buffer + strlen(key) + 2;
+    if (s) {
+        while (*s == '\'')
+            s++;
+        while (*++t != '\0');
+        while (--t > s && *t == '\'')
+            *t = '\0';
+    }
+    return s;
+}
+
+/*
+ * Checks for the existance of .install_type file and reads it
+ */
+void analytics_get_install_type(void)
+{
+    char *install_type_filename;
+    analytics_set_data_str(&analytics_data.netdata_install_type, "");
+    analytics_set_data_str(&analytics_data.netdata_prebuilt_distro, "");
+
+    int install_type_filename_len = (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 3);
+    install_type_filename = mallocz(sizeof(char) * install_type_filename_len);
+    snprintfz(install_type_filename, install_type_filename_len - 1, "%s/%s", netdata_configured_user_config_dir, ".install-type");
+    if (unlikely(access(install_type_filename, R_OK) != 0)) {
+        freez(install_type_filename);
+        return;
+    }
+
+    FILE *fp = fopen(install_type_filename, "r");
+    if (fp) {
+        char *s, buf[256 + 1];
+        size_t len = 0;
+
+        while ((s = fgets_trim_len(buf, 4096, fp, &len))) {
```

Comment:
We are passing as argument `4096`, but `buf` has only 256 bytes length. Whether you see the source code [here](https://github.com/netdata/netdata/blob/714d5a0fccf839ef4a7ff5401430fdca02f97300/libnetdata/libnetdata.c#L1081), you will verify possible problems, please, adjust the argument here or `buf` size.

Reply:
Embarrassing!...